src/libraries/System.Drawing.Common/src/System/Drawing/Gdiplus.cs

```diff
@@ -42,21 +45,7 @@ static Gdip()
             /// a per-thread basis. This way we can avoid 'object in use' crashes when different threads are
             /// referencing the same drawing object.
             /// </summary>
-            internal static IDictionary ThreadData
-            {
-                get
-                {
-                    LocalDataStoreSlot slot = Thread.GetNamedDataSlot(ThreadDataSlotName);
-                    IDictionary? threadData = (IDictionary?)Thread.GetData(slot);
-                    if (threadData == null)
-                    {
-                        threadData = new Hashtable();
-                        Thread.SetData(slot, threadData);
-                    }
-
-                    return threadData;
-                }
-            }
+            internal static IDictionary ThreadData => t_threadData ??= new Hashtable();
```

Comment:
Curious, why did you change this?

Reply:
`[ThreadStatic]` is faster and simpler, and named data slots are nowhere used in the BCL (the shared framework at least). I was originally going to add this commit to another PR that will come after this one, but moved it here.

Reply:
@stephentoub would you mind lending me your pair of eyes to make sure this change is fine?

Reply:
Looks fine, thanks.